tests/scripts/mbedtls_test.py

```diff
@@ -0,0 +1,273 @@
+
+    @staticmethod
+    def align_32bit(b):
+        """
+        4 byte aligns byte array.
+
+        :return:
+        """
+        b += bytearray((4 - (len(b))) % 4)
+
+    @staticmethod
+    def hex_str_bytes(hex_str):
+        """
+        Converts Hex string representation to byte array
+
+        :param hex_str:
+        :return:
+        """
+        assert hex_str[0] == '"' and hex_str[len(hex_str) - 1] == '"', \
+            "HEX test parameter missing '\"': %s" % hex_str
+        hex_str = hex_str.strip('"')
+        assert len(hex_str) % 2 == 0, "HEX parameter len should be mod of 2: %s" % hex_str
+        b = bytearray()
+
+        for i in xrange(len(hex_str) / 2):
+            h = hex_str[i * 2] + hex_str[(i * 2) + 1]
+            try:
+                b += bytearray([int(h, 16)])
+            except ValueError:
+                raise ValueError("Invalid HEX value: %s" % hex_str)
+        return b
+
+    def parameters_to_bytes(self, b, parameters):
+        for typ, param in parameters:
+            if typ == 'int' or typ == 'exp':
+                i = int(param)
+                b += 'I' if typ == 'int' else 'E'
+                self.align_32bit(b)
+                b += bytearray([((i >> x) & 0xff) for x in [24, 16, 8, 0]])
+            elif typ == 'char*':
+                param = param.strip('"')
+                i = len(param) + 1  # + 1 for null termination
+                b += 'S'
+                self.align_32bit(b)
+                b += bytearray([((i >> x) & 0xff) for x in [24, 16, 8, 0]])
+                b += bytearray(list(param))
+                b += '\0'   # Null terminate
+            elif typ == 'hex':
+                hb = self.hex_str_bytes(param)
+                b += 'H'
+                self.align_32bit(b)
+                i = len(hb)
+                b += bytearray([((i >> x) & 0xff) for x in [24, 16, 8, 0]])
+                b += hb
+        return b
+
+    def run_next_test(self):
+        """
+        Send next test function to the target.
+
+        """
+        self.test_index += 1
+        self.dep_index = 0
+        if self.test_index < len(self.tests):
+            name, function, deps, args = self.tests[self.test_index]
+            self.log("Running: %s" % name)
+            bytes = bytearray([len(deps)])
+            if len(deps):
+                bytes += bytearray(deps)
+            bytes += bytearray([function, len(args)])
+            self.parameters_to_bytes(bytes, args)
+            key = bytearray([((len(bytes) >> x) & 0xff) for x in [24, 16, 8, 0]])
+            #self.log("Bytes: " + " ".join(["%x '%c'" % (x, x) for x in bytes]))
+            self.send_kv(key, bytes)
+        else:
+            self.notify_complete(True)
+
+    @staticmethod
+    def get_result(value):
+        try:
+            return int(value)
+        except ValueError:
+            ValueError("Result should return error number. Instead received %s" % value)
+        return 0
+
+    @event_callback('GO')
```

Comment:
The callback flow is actually forced by htrun. That is tool that receives k,v pairs from target and dispatches them into this class as events.